File: Day319/SumString.java
//{ Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            String S = read.readLine();

            Solution ob = new Solution();
            System.out.println(ob.isSumString(S));
        }
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution{
    public int isSumString(String S){
        // code here
        for (int i=1; i<S.length(); i++) {
            for (int j=1; j<S.length()-i; j++) {
                if (find(S, 0, i, j)) {
                    return 1;
                }
            }
        }
        return 0;
    }
    public boolean find(String s, int pos, int i, int j) {
        String temp1 = s.substring(pos, pos+i);
        String temp2 = s.substring(pos+i, pos+j+i);
        String temp3 = sum(temp1, temp2);
        if (temp3.length() > s.length() - pos - i - j) {
            return false;
        }
        if (s.substring(pos+i+j, pos+i+j+temp3.length()).equals(temp3)) {
            if (pos + i + j + temp3.length() == s.length()) {
                return true;
            }
            return find(s, pos+i, j, temp3.length());
        }
        return false;
    }
    public String sum(String s1, String s2) {
        int i = s1.length()-1, j = s2.length()-1;
        StringBuilder answer = new StringBuilder();
        int c = 0;
        while (i >= 0 && j >= 0) {
            int x = s1.charAt(i)-'0';
            int y = s2.charAt(j)-'0';
            int temp = x + y + c;
            int rem = temp%10;
            c = temp/10;
            answer.append(rem);
            i--;
            j--;
        }
        while (i>=0) {
            int x = s1.charAt(i)-'0';
            int temp = x + c;
            int rem = temp%10;
            c = temp/10;
            answer.append(rem);
            i--;
        }
        while (j>=0) {
            int x = s2.charAt(j)-'0';
            int temp = x + c;
            int rem = temp%10;
            c = temp/10;
            answer.append(rem);
            j--;
        }
        while (c>0) {
            int rem = c % 10;
            answer.append(rem);
            c/=10;
        }
        return answer.reverse().toString();
    }
}

File: Day319/SumString.go
package main

import (
	"fmt"
	"strconv"
)

func isSumString(s string) int {
	for i := 1; i < len(s); i++ {
		for j := 1; j < len(s)-i; j++ {
			if find(s, 0, i, j) {
				return 1
			}
		}
	}
	return 0
}

func find(s string, pos, i, j int) bool {
	temp1 := s[pos : pos+i]
	temp2 := s[pos+i : pos+j+i]
	temp3 := sum(temp1, temp2)

	if len(temp3) > len(s)-pos-i-j {
		return false
	}

	if s[pos+i+j:pos+i+j+len(temp3)] == temp3 {
		if pos+i+j+len(temp3) == len(s) {
			return true
		}
		return find(s, pos+i, j, len(temp3))
	}

	return false
}

func sum(s1, s2 string) string {
	i, j := len(s1)-1, len(s2)-1
	var answer string
	c := 0

	for i >= 0 && j >= 0 {
		x, _ := strconv.Atoi(string(s1[i]))
		y, _ := strconv.Atoi(string(s2[j]))
		temp := x + y + c
		rem := temp % 10
		c = temp / 10
		answer = strconv.Itoa(rem) + answer
		i--
		j--
	}

	for i >= 0 {
		x, _ := strconv.Atoi(string(s1[i]))
		temp := x + c
		rem := temp % 10
		c = temp / 10
		answer = strconv.Itoa(rem) + answer
		i--
	}

	for j >= 0 {
		x, _ := strconv.Atoi(string(s2[j]))
		temp := x + c
		rem := temp % 10
		c = temp / 10
		answer = strconv.Itoa(rem) + answer
		j--
	}

	for c > 0 {
		rem := c % 10
		answer = strconv.Itoa(rem) + answer
		c /= 10
	}

	return answer
}

func main() {
	var t int
	fmt.Scan(&t)

	for i := 0; i < t; i++ {
		var s string
		fmt.Scan(&s)

		result := isSumString(s)
		fmt.Println(result)
	}
}
